test/test-multi-app-routes.js

```diff
@@ -0,0 +1,70 @@
+var helper = require('./helper');
+var fs = require('fs');
+var path = require('path');
+
+helper('Test multi-app REST API behavior', function(t, app, baseDir) {
+  t.test('add endpoints', function(tt) {
+    tt.plan(11);
+    app._nginxCmd = function(action, cmdCb) {
+      tt.equal(action, 'reload');
+      if (cmdCb) return cmdCb();
+    };
+
+    var Config = app.models.Config;
+    var Endpoint = app.models.Endpoint;
+
+    Config.setEndpoints([
+        {host: '127.0.0.1', port: '5000', serviceId: 1},
+        {host: '127.0.0.2', port: '5001', serviceId: 2}
+      ], function(err) {
+        tt.ifError(err);
+
+        Endpoint.find({}, function(err, endpoints) {
+          tt.ifError(err);
+          tt.equal(endpoints.length, 2);
+          tt.equal(endpoints[0].host, '127.0.0.1');
+          tt.equal(endpoints[1].host, '127.0.0.2');
+
+          var fdata = fs.readFileSync(path.resolve(baseDir,
+              './nginx/nginx.conf'), 'utf8');
+
+          tt.ok(!!fdata.match(/listen 0.0.0.0:0;/));
+          tt.ok(!!fdata.match(/upstream svc_1/));
+          tt.ok(!!fdata.match(/server 127.0.0.2:5001;/));
+          tt.ok(!!fdata.match(/upstream svc_2/));
+          tt.ok(!!fdata.match(/server 127.0.0.2:5001;/));
+
+          tt.end();
+        });
+      });
+  });
+
+  t.test('remove endpoints', function(tt) {
+    tt.plan(8);
+    app._nginxCmd = function(action, cmdCb) {
+      tt.equal(action, 'reload');
+      if (cmdCb) return cmdCb();
+    };
+
+    var Config = app.models.Config;
+    var Endpoint = app.models.Endpoint;
+
+    Config.setEndpoints([], function(err) {
+      tt.ifError(err);
+
+      Endpoint.find({}, function(err, endpoints) {
+        tt.ifError(err);
+        tt.equal(endpoints.length, 0);
+
+        var fdata = fs.readFileSync(path.resolve(baseDir,
+            './nginx/nginx.conf'), 'utf8');
+
+        tt.ok(!!fdata.match(/listen 0.0.0.0:0;/));
```

Comment:
Cleaner as
```js
tt.match(fdata, /listen 0.0.0.0:0;/);
```

The real advantage is when it fails you get a _far_ more useful failure message.